pkg/ipamd/datastore/data_store.go

```diff
@@ -872,8 +1078,30 @@ func (ds *DataStore) GetENIs() int {
 	return len(ds.eniPool)
 }
 
-// GetENIIPs returns the known (allocated & unallocated) ENI IPs.
-func (ds *DataStore) GetENIIPs(eniID string) ([]string, error) {
+// GetENICIDRs returns the known (allocated & unallocated) ENI secondary IPs and Prefixes
+func (ds *DataStore) GetENICIDRs(eniID string) ([]string, []string, error) {
+	ds.lock.Lock()
+	defer ds.lock.Unlock()
+
+	eni, ok := ds.eniPool[eniID]
+	if !ok {
+		return nil, nil, errors.New(UnknownENIError)
+	}
+
+	var ipPool []string
+	var prefixPool []string
+	for _, assignedAddr := range eni.AvailableIPv4Cidrs {
+		if !assignedAddr.IsPrefix {
+			ipPool = append(ipPool, assignedAddr.Cidr.IP.String())
+		} else {
+			prefixPool = append(prefixPool, assignedAddr.Cidr.String())
+		}
+	}
+	return ipPool, prefixPool, nil
+}
+
+// GetENIPrefixes returns the known (allocated & unallocated) ENI Prefixed.
+func (ds *DataStore) GetENIPrefixes(eniID string) ([]string, error) {
```

Comment:
why we have this given we already have `GetENICIDRs`

Reply:
Yes this again has come from merge, I am not using this anymore.